### PR TITLE
CD-40 footer

### PR DESCRIPTION
**Pull request by left23**
### Summary

1. add wrapper elements with classes for styling
2. add mandate text,
3. replace dummy footer menu items with real
4. replace copyright notice with real and hyperlink
5. renamed 'provided-by' component to 'mandate' and dropped the footer prefix
6. remove cssgrid layout for footer elements, in favour of flexbox
7. menu expands to fill available space
8. social media expands based on content
9. mandate has max-width, but shrinks to fit next to copyright block on tablet, and stacks on mobile
10. copyright has max-width, stacks on mobile
11. added child selector for footer menu so it works better in Drupal without having to add classes to each li
12. added soft footer element and some basic nested rules to control child item layout (this component is optional)
13. changed the grid row of the footer to accommodate for the soft footer above it

### To test:
pull PR and compile styleguide
View in viewports of various sizes

@rupl I tested in IE11 and was happy with the results so the autoprefixer issue I mentioned is irrelevant.

### CODE REVIEW

File: common-design/css/styles.css [hunk 1]
```diff
@@ -935,17 +936,18 @@ button.cd-user-menu__item {
...
-    display: grid;
-    -ms-grid-columns: 1fr 12px 1fr;
-    -ms-grid-rows: auto 48px auto 48px auto;
-    grid-template-columns: repeat(2, 1fr);
-    grid-gap: 12px;
-    grid-row-gap: 48px; }
+    display: -webkit-flex;
+    display: flex;
+    -webkit-flex-wrap: wrap;
+    flex-wrap: wrap;
```
Comment (left23) [inline common-design/css/styles.css:943]: @rupl I notice I am not getting any -ms- prefixes for flexbox. I believe -ms-grid etc was manually added in the previous version

Comment (rupl) [reply common-design/css/styles.css:943]: if autoprefixer decides it's not necessary, then we believe autoprefixer 😄 — the compatibility library is based on caniuse data which is potentially updated on anytime `npm i` is run. So the results of autoprefixer can change even when it seems like nothing changed.

**Comment by left23** [comment]
@rupl Please review when you have a chance.
You will find screenshots on the [issue](https://humanitarian.atlassian.net/browse/CD-40)

I think it's better to replace the social media icons in another PR as this already has a lot going on.

**Comment by rupl** [comment]
When I view the CD Demo, it all looks great. But When I view the front page of the website I am not seeing logos or other pieces that show on the current homepage: https://un-ocha.github.io/styleguide/

Couple screenshots below. Possible that I skipped a step? I ran `gulp dev` and opened my browser to check it out.

sg-frontpage-lg
sg-frontpage-sm

**Comment by left23** [comment]
@rupl 
Ah, no, it's me who overlooked this. I've paid no attention to the actual styleguide part, just the demo (oops)
I'll look after it soon. Thanks!

**Comment by rupl** [comment]
This is just one more reason we should have a single source of truth for these components. It's a bit strange that the demo/front page aren't consuming the same components. Would be nice to set up pages with specific variants like http://demo.patternlab.io/?p=pages-homepage

**Comment by rupl** [comment]
As mentioned in #21 one side effect of merging that PR is that you'll fix the footer issue I identified above.

**Pull request merged**